satellite/satellitedb/mutex.go

```diff
@@ -0,0 +1,267 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package satellitedb
+
+import (
+	"context"
+	"errors"
+	"sync"
+	"time"
+
+	"storj.io/storj/pkg/accounting"
+	"storj.io/storj/pkg/bwagreement"
+	"storj.io/storj/pkg/datarepair/irreparable"
+	"storj.io/storj/pkg/datarepair/queue"
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/pkg/statdb"
+	"storj.io/storj/pkg/storj"
+	"storj.io/storj/storage"
+)
+
+// Mutex wraps DB in a mutex for non-concurrent databases
+type Mutex struct {
+	mu sync.Mutex
+	db *DB
```

Comment:
Nit: consider naming it something different than `db` for less confusion later on

Reply:
Any suggestions?

Reply:
fixed